pkg/validations/base.go

```diff
@@ -46,11 +50,20 @@ func RunValidations(request reconcile.Request, obj client.Object, kind string, i
 	engine.ClearMetrics(result.Reports, promLabels)
 
 	for _, report := range result.Reports {
+		check, ok := engine.registeredChecks[report.Check]
```

Comment:
This should be a function.  engine.GetCheckByName or something.  The fact that the engine stores this as a map should be transparent external to the engine.

Reply:
https://github.com/app-sre/deployment-validation-operator/blob/541effb687e5b5435c76b3eec278e7eb1d786436/pkg/validations/base.go#L55-L59

https://github.com/app-sre/deployment-validation-operator/blob/541effb687e5b5435c76b3eec278e7eb1d786436/pkg/validations/validation_engine.go#L174-L180

Done